source/common/http/header_utility.cc

```diff
@@ -235,7 +235,96 @@ bool HeaderUtility::headerNameContainsUnderscore(const absl::string_view header_
   return header_name.find('_') != absl::string_view::npos;
 }
 
+namespace {
+// This function is used to validate the authority as is done in nghttp2 (based on
+// https://github.com/nghttp2/nghttp2/blob/d97bc7d8745ded136efa6e9e747f2310406893dd/lib/nghttp2_helper.c#L752).
+// Specifically, it permits the character "@" which is not permitted by oghttp2's
+// implementation. Note that this function is used for both H/1 and H/2.
+// The H/1 spec allows "user-info@host:port" for the authority, and the H/2 spec doesn't.
+// Once UHV is used, this function should be removed.
+bool compliant_nghttp2_check_authority(const absl::string_view header_value) {
```

Comment:
I'd probably call this `validate_http1_authority()`.

Reply:
Rephrased to a different name.
IMHO the validation should probably be more rigorous (e.g., up to a single "@" and ":" allowed, etc.), so I've decided to keep the h1_h2 as part of the name.
That said, I don't feel strongly about this, so let me know if you think otherwise.